processing/src/main/java/io/druid/query/aggregation/cardinality/CardinalityAggregatorFactory.java

```diff
@@ -94,11 +103,102 @@ public static Object estimateCardinality(Object object)
 
   private static final byte CACHE_TYPE_ID = (byte) 0x8;
   private static final byte CACHE_KEY_SEPARATOR = (byte) 0xFF;
+  private static final CardinalityAggregatorTypeHelperFactory TYPE_HELPER_FACTORY = new CardinalityAggregatorTypeHelperFactory();
+
 
   private final String name;
   private final List<DimensionSpec> fields;
   private final boolean byRow;
 
+  private static class CardinalityAggregatorTypeHelperFactory implements QueryTypeHelperFactory<CardinalityAggregatorTypeHelper>
```

Comment:
I've moved this to a top-level class to shorten the class name when used by other classes